Sources/NIONFS3/NFSTypes+Containers.swift

```diff
@@ -0,0 +1,529 @@
+        self.writeInteger(rpcNFSReply.rpcReply.xid, endianness: .big)
+
+        self.writeRPCReply(rpcNFSReply.rpcReply)
+
+        switch rpcNFSReply.nfsReply {
+        case .mountNull:
+            () // noop
+        case .mount(let nfsReplyMount):
+            self.writeNFSReplyMount(nfsReplyMount)
+        case .unmount(let nfsReplyUnmount):
+            self.writeNFSReplyUnmount(nfsReplyUnmount)
+        case .null:
+            () // noop
+        case .getattr(let nfsReplyGetAttr):
+            self.writeNFSReplyGetAttr(nfsReplyGetAttr)
+        case .fsinfo(let nfsReplyFSInfo):
+            self.writeNFSReplyFSInfo(nfsReplyFSInfo)
+        case .pathconf(let nfsReplyPathConf):
+            self.writeNFSReplyPathConf(nfsReplyPathConf)
+        case .fsstat(let nfsReplyFSStat):
+            self.writeNFSReplyFSStat(nfsReplyFSStat)
+        case .access(let nfsReplyAccess):
+            self.writeNFSReplyAccess(nfsReplyAccess)
+        case .lookup(let nfsReplyLookup):
+            self.writeNFSReplyLookup(nfsReplyLookup)
+        case .readdirplus(let nfsReplyReadDirPlus):
+            self.writeNFSReplyReadDirPlus(nfsReplyReadDirPlus)
+        case .readdir(let nfsReplyReadDir):
+            self.writeNFSReplyReadDir(nfsReplyReadDir)
+        case .read(let nfsReplyRead):
+            nextStep = self.writeNFSReplyReadPartially(nfsReplyRead)
+        case .readlink(let nfsReplyReadlink):
+            self.writeNFSReplyReadlink(nfsReplyReadlink)
+        case .setattr(let nfsReplySetattr):
+            self.writeNFSReplySetattr(nfsReplySetattr)
+        case ._PLEASE_DO_NOT_EXHAUSTIVELY_MATCH_THIS_ENUM_NEW_CASES_MIGHT_BE_ADDED_IN_THE_FUTURE:
+            // inside the module, matching exhaustively is okay
+            preconditionFailure("unknown NFS3 reply, this should never happen. Please report a bug.")
+        }
+
+        self.setRPCFragmentHeader(.init(length: UInt32(self.writerIndex - startWriterIndex - 4 + nextStep.bytesToFollow),
+                                        last: true),
+                                  at: startWriterIndex)
+        return (self.writerIndex - startWriterIndex, nextStep)
+    }
+
+    @discardableResult
+    public mutating func writeRPCNFSReply(_ reply: RPCNFS3Reply) -> Int {
+        let (bytesWritten, nextStep) = self.writeRPCNFSReplyPartially(reply)
+        switch nextStep {
+        case .doNothing:
+            return bytesWritten
+        case .writeBlob(let buffer, numberOfFillBytes: let fillBytes):
+            return bytesWritten
+            &+ self.writeImmutableBuffer(buffer)
+            &+ self.writeRepeatingByte(0x41, count: fillBytes)
+        }
+    }
+
+    public mutating func readRPCMessage() throws -> (RPCMessage, ByteBuffer)? {
+        let save = self
+        guard let fragmentHeader = try self.readRPCFragmentHeader(),
+              let xid = self.readInteger(endianness: .big, as: UInt32.self),
```

Comment:
```suggestion
              let xid = self.readInteger(as: UInt32.self),
```

Reply:
@Lukasa I though explicit over implicit, but can remove. wdyt?

Reply:
I honestly read it as just noise. Big-endian is overwhelmingly the default, so I don't tend to need anyone to tell me that. I _do_ need them to tell me when they're doing something unusual though.

Reply:
ok

Reply:
done